src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCauseBuildAction.java

```diff
@@ -342,7 +343,17 @@ private static void adjustProjectDisplayName(
     private static Set<Run<?, ?>> getDownstreamBuilds(
             final Run build) {
 
-        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>();
+        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>(new Comparator<Run<?, ?>>() {
+            @Override
+            public int compare(Run<?, ?> o1, Run<?, ?> o2) {
+                final int res = o1.getParent().getFullName().compareTo(o2.getParent().getFullName());
```

Comment:
@rsandell I see no risk of NPE here. If you see the code where `parent` may be really null, please point me to it